numpy/core/src/multiarray/item_selection.c

```diff
@@ -1749,6 +1750,281 @@ PyArray_LexSort(PyObject *sort_keys, int axis)
     return NULL;
 }
 
+/*
+* Helper function for PyArray_KeySort
+* Sorts 1D slice based on argsort indices
+* takes in raw bytes of data, indices array
+* Number of elements, and stride of data
+* element size and temporary storage for an element.
+*/
+void reorder_by_index(char *data, char *current, npy_intp *index, npy_intp N, npy_intp stride, npy_intp elsize) {
+    // Reorder the indices and relevant values cyclically
+    for(npy_intp i = 0; i < N; i++){
+        /*Backup current element, end of loop place it in correct location.
+        *Stride here handles both aligned and non-aligned case
+        *When stride is same as elsize we are operating on aligned data.
```

Comment:
You actually mean "contiguous" here and not aligned (although aligned is also true). I think you can actually just remove the stride part of the comment, stride always means the same thing in the NumPy code base.